File: backend/Models/Order.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class Order : BaseModel
    {
        public Order() { }

        public Order(string customerEmail, ShippingAddress shippingAddress, DeliveryMethod deliveryMethod, IReadOnlyList<OrderItem> orderItems, decimal subTotal)
        {
            CustomerEmail = customerEmail;
            ShippingAddress = shippingAddress;
            DeliveryMethod = deliveryMethod;
            OrderItems = orderItems;
            SubTotal = subTotal;
        }

        public string CustomerEmail { get; set; }
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
        public ShippingAddress ShippingAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal SubTotal { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentMethodId { get; set; }

        public decimal GetTotal() => SubTotal + DeliveryMethod.Price;
    }
}

File: frontend/src/app/models/order.ts
import { Address } from './address';
import { OrderItem } from './order-item';

export interface Order {
  id: number;
  customerEmail: string;
  orderDate: Date;
  shippingAddress: Address;
  deliveryMethodPrice: number;
  deliveryMethod: string;
  orderItems: OrderItem[];
  subTotal: number;
  orderStatus: string;
  total: number;
  paymentMethodId: string;
}
